metadata/src/main/java/org/apache/kafka/controller/ConfigurationControlManager.java

```diff
@@ -122,19 +126,13 @@ private void incrementalAlterConfigResource(ConfigResource configResource,
                     newValue = opValue;
                     break;
                 case DELETE:
-                    if (opValue != null) {
```

Comment:
Do we have any cases where a value is provided for a DELETE operation or was this just to make existing tests work?

Reply:
There are existing integration tests that pass a non-null value. This PR doesn't convert those tests over but a future one will.